src/generic-container.ts

```diff
@@ -239,11 +246,28 @@ export class GenericContainer implements TestContainer {
     return this;
   }
 
+  public withCopyFileToContainer(sourcePath: string, containerPath: string): this {
+    this.getTarToCopy().file(sourcePath, { name: containerPath });
+    return this;
+  }
+
+  public withCopyContentToContainer(content: string | Buffer | Readable, containerPath: string): this {
+    this.getTarToCopy().append(content, { name: containerPath });
+    return this;
+  }
+
   public async isImageCached(dockerClient: DockerClient): Promise<boolean> {
     const dockerImageNames = await dockerClient.fetchDockerImageNames();
     return dockerImageNames.some((dockerImageName) => dockerImageName.equals(this.dockerImageName));
   }
 
+  protected getTarToCopy(): archiver.Archiver {
```

Comment:
Any reason to make this protected or could it be private?

Reply:
Well, the `tarToCopy` field itself is protected (in line with all the other fields), and if some subclass intends to use it, then I think they should also have access to this getter which also acts as the lazy initializer for the field.